tools/affiliations/importer/src/main.rs

```diff
@@ -1,17 +1,369 @@
-        Ok(names) => names.iter().for_each(|n| println!("{}", n)),
-        Err(error) => panic!("{:#?}", error),
+    let logins = crawler
+        .list_github_logins()
+        .await
+        .expect("Failed to list logins.");
+    let mut logins: HashSet<&String> = logins.iter().collect::<HashSet<_>>();
+    info!("Fetching the logins to succeed!");
+
+    for record in &mut records {
+        if logins.contains(&record.login) {
+            set_pingcap_affiliation(record);
+            logins.remove(&record.login);
+        } else {
+            remove_pingcap_affiliation(record);
+        }
+    }
+
+    logins.iter().for_each(|l| {
+        info!("Add new record for {}.", l);
+        records.push(GitHubUser {
+            login: (*l).clone(),
+            email: l.as_str().to_owned() + "!users.noreply.github.com",
+            affiliation: Some(PINGCAP_AFFILIATION.to_string()),
+            source: Some("manual".to_string()),
+            name: None,
+            commits: 0,
+            location: None,
+            country_id: None,
+        })
+    });
+
+    let file =
+        std::fs::File::create("github_users.json").expect("Create github_users file failed.");
+    serde_json::to_writer_pretty(file, &records).expect("Failed to deserialize.");
+}
+
+/// Set PingCAP affiliation to user.
+fn set_pingcap_affiliation(user: &mut GitHubUser) {
+    match &user.affiliation {
+        // If none, set directly to PingCAP.
+        None => {
+            user.affiliation = {
+                info!("Set [{}] affiliation to PingCAP.", user.login);
+                Some(PINGCAP_AFFILIATION.to_string())
+            }
+        }
+        Some(affiliation) => {
+            let new_affiliation = generate_new_affiliation_with_pingcap(affiliation);
+            match new_affiliation {
+                None => {}
+                Some(new_affiliation) => {
+                    info!(
+                        "Set [{}] affiliation from `{}` to `{}`.",
+                        user.login, affiliation, new_affiliation
+                    );
+                    user.affiliation = Some(new_affiliation);
+                }
+            }
+        }
+    }
+}
+
+/// Remove PingCAP affiliation to user.
+fn remove_pingcap_affiliation(user: &mut GitHubUser) {
+    match &user.affiliation {
+        // None, no need to change.
+        None => {}
+        Some(affiliation) => {
+            // Only when the affiliation is valid.
+            if !INVALID_AFFILIATIONS.contains(&affiliation.as_str()) {
+                match affiliation
+                    .split(COMPANY_SEPARATOR)
+                    .collect::<Vec<&str>>()
+                    .last()
+                {
+                    None => {}
+                    Some(last) => {
+                        // Only if the last company is PingCAP.
+                        if *last == PINGCAP_AFFILIATION {
+                            let current_date = Utc::now().date().format("%Y-%m-%d").to_string();
```

Comment:
@Mini256 here

Reply:
But in fact, we didn't change the upstream JSON.

For example:
If we updated the affiliation information on May 15, The output is "PingCAP <2021-05-15, independent".

When updating the affiliation information while June 15, climbing JSON gets from the upstream JSON is still "PingCAP ", which will become "PingCAP <2021-06-15, independent" after remove_pingcap_affiliation processing.

This actor information may always be subordinate to PingCAP.